src/sentry/rules/processing/buffer_processing.py

```diff
@@ -183,14 +194,10 @@ def process_buffer() -> None:
 
             project_ids = list(all_project_ids_and_timestamps.keys())
             for project_id in project_ids:
-                process_in_batches(project_id, processing_type)
+                process_in_batches(buffer, project_id, processing_type)
 
-            buffer.backend.delete_keys(
+            buffer.delete_keys(
                 buffer_keys,
                 min=0,
                 max=fetch_time,
             )
-
-
-if not redis_buffer_registry.has(BufferHookEvent.FLUSH):
```

Comment:
🎉 - i never liked this code. just curious, if we want to add another handler when the buffer is flushed, would we have to manually add it to `tasks/process_buffer.py::process_pending_batch`? should we have a registry hook there or is this uncommon enough that we can be explicit about it?